barman/recovery_executor.py

```diff
@@ -1807,13 +1807,94 @@ def _backup_copy(self, backup_info, dest, remote_command=None, **kwargs):
             raise DataTransferFailure.from_command_error("rsync", e, msg)
 
 
+class IncrementalRecoveryExecutor(RemoteConfigRecoveryExecutor):
+    """
+    A specialised recovery method for incremental backups.
+    Inheritence is not necessarily the best thing here since the two RecoveryExecutor
+    classes only differ by this one method, and the same will be true for future
+    RecoveryExecutors (i.e. ones which handle encryption).
+    Nevertheless for a wip "make it work" effort this will do.
+    """
+
+    BASE_INCREMENTAL_NAME = "base"
+
+    def __init__(self, backup_manager):
+        """
+        Constructor
+
+        :param barman.backup.BackupManager backup_manager: the BackupManager
+            owner of the executor
+        :param compression Compression.
+        """
+        super(IncrementalRecoveryExecutor, self).__init__(backup_manager)
+
+    # TODO
+    # When recovering, we need to:
+    # 1. create the backup chain of parents
+    # 2. combine backups into a single synthetic backup
+    # 3. Check if the output file exists
+    # 3. temporarily store this in a folder from barman server
+    # 4. copy the backup to a node
+
+    def get_backup_chain(self, backup_info):
+        """
+        Get the backup chain for the provided incremental backup, returning it as a list
+        starting from the full backup, through the parent backups up until the provided
+        incremental backup.
+
+        :param backup_info: The backup_info object for the incremental backup.
+        :return: A list of backup_info objects from the root to the provided incremental backup.
+        """
+        # Convert the generator to a list and reverse it to start from the root backup
+        return [
+            backup
+            for backup in reversed(list(backup_info.walk_to_root(include_self=True)))
```

Comment:
check [dev/bar-168](https://github.com/EnterpriseDB/barman/pull/950/files#diff-458020cb725a00dcf8f7c7a3902a83a1aa05ff62d48dea5e1e1442cf3ad9e40bR141 ). Maybe there is a way that you do not need to turn the generator into the list. Maybe find a way to run a for loop and insert each value in the beginning of a list instead of creating the list first and then reversing it.